### ISSUE TITLE
So many duplications.

### ISSUE THREAD

**Issue by paul-bio** [chhylp123/hifiasm#292]
Hello, I recently performed a de novo genome assembly using HiFiasm.
And I have Hifi sequencing data.

First, thank you let us use this wonderful tool.
But when i run for the first time,
I got lot of duplicated genes in **dip.p_ctg.fa** _(seems diploid  assembled file)_

here is the results I got from BUSCO analysis.

For **hap1.p_ctg.fa**
![image](https://user-images.githubusercontent.com/73869977/176891483-c814309c-6a2c-4e58-ba48-5e6ec71f9648.png)
with contigs number of 22,298

For **hap2.p_ctg.fa**
![image](https://user-images.githubusercontent.com/73869977/176891536-001fc9f7-d69d-427f-8537-fac98d1e498c.png)
with contigs number of 16,918

For **dip.p_ctg.fa**
![image](https://user-images.githubusercontent.com/73869977/176891602-a9c31d33-7d44-4218-aa33-3451d72d728e.png)
with contigs number of 39,216

It seems like genes are lost in **hap1.p_ctg.fa** but when i look up the **dip.p_ctg.fa**, I got reliable result.

So I run a purge_hapolotig with **dip.p_ctg.fa** file, and got like this.
![image](https://user-images.githubusercontent.com/73869977/176892621-3987cefe-300e-4ed0-be20-ce6cab4add64.png)

Do you have any suggestion?
It would be best when I use result from **dip.p_ctg.fa**...

Thanks anyway.

**Comment by chhylp123** [comment]
Are these assemblies much larger than the estimated genome size?

**Comment by paul-bio** [comment]
Hi @chhylp123 

It seems this species has **18Gb** of genome size.


Here are the summary of genome result.




For **hap1.p_ctg.fa**
![image](https://user-images.githubusercontent.com/73869977/176981505-78105a18-9567-4d7e-b57d-12fe2fe5435a.png)



For **hap2.p_ctg.fa**
![image](https://user-images.githubusercontent.com/73869977/176981519-4934f84b-91e8-4dc9-b8a9-f2073be372a4.png)


And for **dip.p_ctg.fa**
![image](https://user-images.githubusercontent.com/73869977/176981526-4e11bc80-c2f3-4320-b115-9ad50e05c448.png)

**Comment by chhylp123** [comment]
Looks like the coverage information was inferred incorrectly by hifiasm. Could you please have a try with the methods listed here (https://hifiasm.readthedocs.io/en/latest/faq.html#why-the-size-of-primary-assembly-or-partially-phased-assembly-is-much-larger-than-the-estimated-genome-size)?

**Comment by paul-bio** [comment]
Thanks @chhylp123 .

I will rerun hifiasm with two ways (-s 0.5 and -s 0.45)

But since this species has highly heterozygosity genome, should I change -l parameters aswell?

And can you tell me what is the default of -l ?


Thanks again.
From Paul.

**Comment by paul-bio** [comment]
Meantime, I tried twice with -s 5 -s0.45 and here is the result

![image](https://user-images.githubusercontent.com/73869977/177922752-a2b64307-03d2-4c81-89bd-efbf692c2e0f.png)

When running with -s 0.45, I got haplotigs which were relatively in same size.But 
This species is expected to have 1.8~1.9Gb of genome size.
and it seems not much duplications are murged enough when compare diploid stats with haploid stat.

In this case, should I run with -l 3 ??

**Comment by chhylp123** [comment]
Sorry for the late reply. What if you have a try with https://github.com/dfguan/purge_dups? The default value for `-l` is `-l3`.

**Comment by paul-bio** [comment]
Previously you mentioned, default value for -l is 1
![image](https://user-images.githubusercontent.com/73869977/178861236-2c538bdb-fb23-4a48-8f23-b06ab6295857.png)

It is bit confusing,, is the default value for -l is -l1 or -l3?

Thanks for reply @chhylp123

**Comment by chhylp123** [comment]
It's my bad. The default value for `-l` is `3`.

**Comment by paul-bio** [comment]
Thanks,

![image](https://user-images.githubusercontent.com/73869977/178862773-4eb3764c-1551-4151-a995-88f79a0481a2.png)


Meantime I run 11 times with different parameters.
And seems it would be best if I purge duplications with p_ctg.fa file (diploid type).

In this case, can i perform **

> purge_dups

** with p_ctg.fa(diploid) rather than hap1.p_ctg.fa(haploid1)?

**Comment by chhylp123** [comment]
Yes. Please note that when you align HiFi reads, it would be better to utilize both `p_ctg.gfa` and `a_ctg.gfa`.

**Comment by paul-bio** [comment]
Thank you so much for your suggestions.

I will try and let you know the results.

**Comment by paul-bio** [comment]
Hello @chhylp123 ,

When using `purge_dups`, I need `alternative assembly hap_asm`.

however alternative files comes when there is --primary option is fed.
In question #243 , you said .p_utg.gfa is a alternative contig file.
Can I use it for  `alternative assembly hap_asm`?


Thanks.

**Comment by chhylp123** [comment]
No, `p_utg.gfa` is the assembly graph. You should use `a_ctg.gfa` as the alternative assembly.

**Comment by paul-bio** [comment]
Hi @chhylp123  there were different results.

When I run hifiams using command below,
`$hifiasm -l1 -s 0.5 --hg-size 1.9g --primary`

I got `*a_ctg.fa` (alternative contigs) and `*p_ctg.fa` (primary contigs) both of which were used for purge_dups.

And I got 
BUSCO values of
C 95, S 34, D 61, F 2.9, M 2.1.


However, I rerun hifiasm without `--primary` options and used `*p_ctg.fa` and `*p_utg.fa` for purge_dups.

And I got
C 93.3, S 78.3, D 15.0, F 3.8, M 2.9.



Is it okay for me to use `*p_utg.fa` as an alternative contig?

**Comment by chhylp123** [comment]
Purge_dups is a little bit tricky to run. There are a lot of issues at the repo of Purge_dups to discuss how to select appropriate parameters for it, so you need to take care of that. In addition, could you check the coordinates of those duplicated genes, and manually filter out some wrong duplications? BUSCO is not such accurate.

**Comment by zhang144999** [comment]
> 
Hello paul

I'm also about to assemble a species with a very large genome recently. This specie own 17G genome. It's sequencing 30× Now the sequencing company only return 2 cells data, it‘s 4×. 
Now, I want to try assembly this species genome. I am worry about my server incompetence.  
The RAM is 1T, 80 threads. 
Now the hifiasm is running, Only this 4× data are using 626G memory, and I am not sure it whether or not continue using memory. I am worry about if I using 30× hifi data, the server it's ok. 
So, Could you tell me how many memory used when you used when you assembly?


Many thx.

**Comment by chhylp123** [comment]
4x coverage might take more RAM than 30x. 1TB should be fine.

**Comment by zhang144999** [comment]
> 
Hi, chhylp123

Does this mean that our servers can assemble this 17G species if all 30x data return.
What about adding hic data? Our hic data is 100x.

Many thx.

**Comment by chhylp123** [comment]
I think it should be fine. Low coverage confuses hifiasm so that it cannot identify the right parameters, making the memory requirement extremely large.
